kudu/src/test/java/org/apache/apex/malhar/kudu/test/KuduClusterAvailabilityTestRule.java

```diff
@@ -0,0 +1,161 @@
+        runThisTest = false;
+      }
+      if ((masterHostsValue == null) && (!testContext.kuduClusterBasedTest())) {
+        runThisTest = true;
+      }
+    }
+    if (runThisTest) {
+      // call the before class equivalent for real kudu cluster
+      if ((testContext != null ) && (testContext.kuduClusterBasedTest())) {
+        try {
+          initRealKuduClusterSetup(masterHostsValue);
+        } catch (Exception e) {
+          LOG.error(e.getMessage(), e);
+        }
+      } else {
+        // call the before class equivalent for mocked kudu cluster
+        try {
+          KuduInputOperatorCommons.initCommonConfigsForAllTypesOfTests();
+        } catch (Exception e) {
+          LOG.error(e.getMessage(),e);
+        }
+      }
+      // Run the original test
+      return base;
+    } else {
+      // bypass the test altogether
+      return new Statement()
+      {
+
+        @Override
+        public void evaluate() throws Throwable
+        {
+          // Return an empty Statement object for those tests
+        }
+      };
+    }
+
+  }
+
+  private void initRealKuduClusterSetup(String masterHosts) throws Exception
+  {
+    if (!KuduClientTestCommons.tableInitialized) {
+      synchronized (KuduClientTestCommons.objectForLocking) { // test rigs can be parallelized
+        if (!KuduClientTestCommons.tableInitialized) {
+          KuduClientTestCommons.setKuduMasterAddresses(masterHosts);
+          KuduClientTestCommons.setup();
+          initPropertiesFilesForBaseKuduOutputOperator(masterHosts);
+          initPropertiesFilesForIncrementalStepScanInputOperator(masterHosts);
+          KuduClientTestCommons.tableInitialized = true;
+        }
+      }
+    }
+  }
+
+  private void initPropertiesFilesForBaseKuduOutputOperator(String masterHosts)
+  {
+    BaseKuduOutputOperator.DEFAULT_CONNECTION_PROPS_FILE_NAME = rewritePropsFile(
+      masterHosts,BaseKuduOutputOperator.DEFAULT_CONNECTION_PROPS_FILE_NAME);
+  }
+
+  private void initPropertiesFilesForIncrementalStepScanInputOperator(String masterHosts)
+  {
+    IncrementalStepScanInputOperator.DEFAULT_CONNECTION_PROPS_FILE_NAME = rewritePropsFile(
+      masterHosts,IncrementalStepScanInputOperator.DEFAULT_CONNECTION_PROPS_FILE_NAME);
+  }
+
+  /**
+   * Creates a temporary properties file that sets the kudu cluster master hosts value as given in the command line.
+   * @param masterHosts
+   * @param originalFileName File name as would be picked by the class loader
+   * @return
+   */
+  private String rewritePropsFile(String masterHosts, String originalFileName)
+  {
+    Properties kuduConnectionProperties = new Properties();
+    ClassLoader loader = Thread.currentThread().getContextClassLoader();
+    InputStream kuduPropsFileAsStream = loader.getResourceAsStream(originalFileName);
+    File tempPropertiesFile = null;
+    try {
+      tempPropertiesFile = File.createTempFile(System.getProperty("java.io.tmpdir"),".properties");
```

Comment:
The first parameter is a temp file prefix, not need to get the system property.

Reply:
Fixed.